server/api/core/ConnectionDataStore.js

```diff
@@ -0,0 +1,42 @@
+import { DDP } from "meteor/ddp-client";
+
+const inMemoryCache = {};
+const connectionKey = "connection-data";
+
+function connectionData() {
+  // similar to Meteor.userId()
+  const invocation =
+    DDP._CurrentMethodInvocation.get() ||
+    DDP._CurrentPublicationInvocation.get();
+
+  if (!invocation) {
+    // operating outside of a connection
+    return inMemoryCache;
+  }
+
+  const { connection } = invocation;
+
+  const hasDataKey = connectionKey in connection;
+
+  if (!hasDataKey) {
+    connection[connectionKey] = {};
+  }
+
+  return connection[connectionKey];
+}
+
+const ConnectionDataStore = {
+  get(key) {
+    return connectionData()[key];
```

Comment:
we want to call `connectionData()` each time we try to access data since `ConnectionDataStore` is a singleton, and we want to be sure we are using the _current_ connection at all times.